prow/cache/types/types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Caches is the basic abstraction over multiple Cache entries
+type Caches []*Cache
+
+// Validate verifies if every Cache contains a name, keys and a path.
+func (c Caches) Validate() error {
+	for _, cache := range c {
+		if cache.Name == "" {
```

Comment:
Do we need enforcement that the Cache Name is unique?

Reply:
Yes, I think I can add it by iterating over all caches and checking for uniqueness. :+1:

Reply:
That or Caches can be a struct with both `[]*Cache` and a `CacheSet := make(map[string]bool)` to have the quick check. But would force a change in all `Caches`